Documentation for adding a field to a field type category in the Content type Manage fields "Add field" interface (#117)
tripal/tripal_doc: docs/dev_guide/module_dev/fields/auto_attach.rst

Add Fields to Content Types
============================

In order for a field to be used, it needs to be added to a specific content type.
A field can also be defined in a YAML file for fields you want to be added automatically, 
see :ref:`Adding a field programatically`.

To add a field manually, this can be done through the "Manage Fields" interface for a given Content Type.

> More specifically, go to **Tripal** → **Page Structure**

Reviewer: I was not able to locate Page Structure as described
<img width="257" height="573" alt="Screenshot 2025-10-10 at 10 43 04 AM" src="https://github.com/user-attachments/assets/1ab8975c-40a5-4662-bcad-38e6b8e124c9" />

But it is in Structure -> Tripal Content Types -> Choose Manage Fields under Operations column of the content type

Author: Oh there are two ways to get to it, I will have to update to make this clearer...

Related new tripal issue https://github.com/tripal/tripal/issues/2323

Author: This should be resolved with commit https://github.com/tripal/tripal_doc/pull/117/commits/b8cd5820020b95ea09a5ee347cb1a4c4dda7b3cb

and then choose "**Manage Fields**" for the Content Type you want to add a field to,
and then click "**+ Create a new field**".

Field Categories
-----------------

Each field is present in one of the categories on this page based on its annotation.

> For Tripal Core fields you will note that we use the categories

Reviewer: ```suggestion
For Tripal Core fields, you will note that we use the categories
```

**Tripal Fields** (`tripal`) and **Chado Fields** (`tripal_chado`),
and this is based on whether the field is defined in the **tripal** or **tripal_chado** module.

  .. image:: auto_attach.1.tripal_field_categories.png
     :width: 711
     :alt: Selectors for Tripal fields and for Chado fields

For extension modules, we recommend determining your category as follows:

 - `tripal`: any generic tripal field which is independent of the storage backend.
   Usually this means it is very application-specific without a biological focus (`e.g.` permission control).

 - `tripal_chado`: any field which is dependent on chado for data storage/retrieval.
   This includes most fields in Tripal Core, and likely most fields in general.

>  - custom category: you may want to make your own category if your field uses a different

Reviewer: ```suggestion
 - `custom category`: you may want to make your own category if your field uses a different
```

   storage backend or REST API. You can also create your own category for a module providing
   a large number of fields supporting specific functionality, but be careful to ensure this
   will be intuitive to your users and keep it functionality focused, not development group
   focused (`e.g.` **Genetic Variants** but not **MyLab**).
   When making a custom category you will also need to provide an icon.
   You can follow the Drupal documentation on how to do this.

As an example, you can see the category `tripal_chado` has been defined for the Additional Type field:

.. code-block:: php

  /**
   * Plugin implementation of Tripal additional type field type.
   */
  #[TripalFieldType(
    id: 'chado_additional_type_type_default',
    category: 'tripal_chado',
    label: new TranslatableMarkup('Chado Type Reference'),
    description: new TranslatableMarkup('A Chado type reference'),
    default_widget: 'chado_additional_type_widget_default',
    default_formatter: 'chado_additional_type_formatter_default',
  )]

>   class ChadoAdditionalTypeTypeDefault extends ChadoFieldItemBase {

Reviewer: was the repeated use 'type' in ChadoAdditionalTypeTypeDefault and id: chado_additional_type_type_default intentional, or could it be a typo?

Author: It is not a typo, it is this way to be consistent with all other chado fields. The "name" is "additional type", so like we have "analysis_type" this then becomes "additional_type_type"

> 
>   /**
>    * Plugin implementation of Tripal additional type field type.
>    */
>   #[TripalFieldType(
>     id: 'chado_additional_type_type_default',
>     category: 'tripal_chado',
>     label: new TranslatableMarkup('Chado Type Reference'),
>     description: new TranslatableMarkup('A Chado type reference'),
>     default_widget: 'chado_additional_type_widget_default',
>     default_formatter: 'chado_additional_type_formatter_default',
>   )]
>   class ChadoAdditionalTypeTypeDefault extends ChadoFieldItemBase {
>   ...

Reviewer: ```suggestion
  use Drupal\Core\StringTranslation\TranslatableMarkup;

  /**
   * Plugin implementation of Tripal additional type field type.
   */
  #[TripalFieldType(
    id: 'chado_additional_type_type_default',
    category: 'tripal_chado',
    label: new TranslatableMarkup('Chado Type Reference'),
    description: new TranslatableMarkup('A Chado type reference'),
    default_widget: 'chado_additional_type_widget_default',
    default_formatter: 'chado_additional_type_formatter_default',
  )]
  class ChadoAdditionalTypeTypeDefault extends ChadoFieldItemBase {
  ...
```

Suggesting the use statement for the reference made in label and description.

For example, this is the first field defined in `tripal_chado/config/install/tripal.tripalfield_collection.general_chado.yml`

.. code-block:: yaml

    -   name: 'organism_genus'
        content_type: 'organism'
        label: 'Genus'
        type: 'chado_string_type_default'
        description: 'The genus name of the organism.'
        cardinality: 1
        required: true
        storage_settings:
            storage_plugin_id: 'chado_storage'
            storage_plugin_settings:
                base_table: 'organism'
                base_column: 'genus'
            max_length: 255
        settings:
            termIdSpace: 'TAXRANK'
            termAccession: '0000005'
        display:
            view:
                default:
                    region: 'content'
                    label: 'above'
                    weight: 10
                    settings:
                      field_prefix: '<em>'
                      field_suffix: '</em>'
            form:
                default:
                    region: 'content'
                    weight: 10

If you need to add a field later, for example in an update hook, code similar to the

> following can be used to load a specific field as defined in a YAML file.

Reviewer: ```suggestion
If you need to add a field later, such as in an update hook, use code similar to the
following to load a specific field as defined in a YAML file.
```

how about?
